src/shared/components/Banners/RebrandBanner/RebrandBanner.tsx

```diff
@@ -0,0 +1,141 @@
+import { Box, Collapse } from '@mui/material';
+import { useEffect, useState } from 'react';
+import { Trans } from 'react-i18next';
+import { useLocation } from 'react-router-dom';
+
+import curiousIcon from 'assets/images/curious_icon--white.png';
+import { auth, workspaces } from 'redux/modules';
+import { theme, variables } from 'shared/styles';
+
+import { Banner, BannerProps } from '../Banner';
+import { StyledImg } from './RebrandBanner.styles';
+
+/**
+ * Returns a unique key for the rebrand banner dismiss state
+ * If workspaceId is provided, it creates a user+workspace specific key
+ * Otherwise, it creates a user-only key for use on the auth screen
+ */
+export const getDismissedKey = (userId: string, workspaceId?: string) =>
+  workspaceId
+    ? `rebrand-banner-dismissed-${userId}:${workspaceId}`
+    : `rebrand-banner-dismissed-${userId}`;
+
+// Global key for anonymous users (e.g., on the login screen)
+export const GLOBAL_DISMISSED_KEY = 'rebrand-banner-dismissed-global';
+
+const DISPLAY_ROUTES = [
+  '/auth',
+  '/dashboard/applets',
+  '/dashboard/managers',
+  '/dashboard/respondents',
+];
+
+export const RebrandBanner = (props: BannerProps) => {
+  const { ownerId } = workspaces.useData() ?? {};
+  const userData = auth.useData();
+  const userId = userData?.user.id;
+  const location = useLocation();
+
+  const [isRebrandBannerActive, setIsRebrandBannerActive] = useState(false);
+  const [isCollapsing, setIsCollapsing] = useState(false);
+
+  const handleOnExited = () => {
+    if (isCollapsing) {
+      setIsCollapsing(false);
+      setIsRebrandBannerActive(false);
+    }
+  };
+
+  useEffect(() => {
+    // For auth screen or when user is not logged in yet
+    if (!userId) {
+      const globalDismissed = localStorage.getItem(GLOBAL_DISMISSED_KEY);
+      setIsRebrandBannerActive(!globalDismissed);
+
+      return;
+    }
+
+    // For logged-in users
+    const userDismissed = localStorage.getItem(getDismissedKey(userId));
+
+    // If there's a workspace, also check workspace-specific dismissal
+    if (ownerId) {
+      const workspaceDismissed = localStorage.getItem(getDismissedKey(userId, ownerId));
+      setIsRebrandBannerActive(!userDismissed && !workspaceDismissed);
+    } else {
+      setIsRebrandBannerActive(!userDismissed);
+    }
+  }, [userId, ownerId]);
+
+  const isDisplayRoute = DISPLAY_ROUTES.some(
+    (route) =>
+      location.pathname === route ||
+      (location.pathname.startsWith(route) && route.includes('/auth')),
+  );
+
+  const handleDismiss = () => {
+    // For auth screen or when user is not logged in yet
+    if (!userId) {
+      localStorage.setItem(GLOBAL_DISMISSED_KEY, 'true');
+    } else {
+      // Set user-level dismissal
+      localStorage.setItem(getDismissedKey(userId), 'true');
```

Comment:
It's tricky to to perfectly align with this AC:

> - If the user dismisses the banner, its dismissed state should be remembered for the duration of the session and across login sessions

One way would be to be a little more aggressive with setting dismissal state, that is, we dismiss the global auth screen banner even when the user is logged in. Else it's a bit annoying when the banner reappears after you've already dismissed it once while being logged in. It seems like the app has "forgotten" your choice.

What do you think?

```suggestion
    // Always dismiss global auth screen banner to prevent banner from reappearing
    // after having already dismissed it during an active user session.
    localStorage.setItem(GLOBAL_DISMISSED_KEY, 'true');

    if (userId) {
      // Set user-level dismissal
      localStorage.setItem(getDismissedKey(userId), 'true');
```

Reply:
On second thought, I'm not so sure my idea is the best. 🙃 The more I think about it the more it feels a bit arbitrary… and the AC doesn't clarify or say to do this explicitly. Maybe just leave it the way you have it, treating the "global" login screen key acting like its own "user" independent of other users.